app/src/vanilla/java/cm/aptoide/pt/view/settings/NewAccountFragment.java

```diff
@@ -0,0 +1,296 @@
+  @Inject NewAccountNavigator newAccountNavigator;
+  private AptoideAccountManager accountManager;
+
+  private Converter.Factory converterFactory;
+  private OkHttpClient httpClient;
+  private BodyInterceptor<BaseBody> bodyInterceptor;
+
+  //Account views
+  private View myProfileView;
+  private View myStoreView;
+  private View loginView;
+  private View accountView;
+  private TextView createStoreMessage;
+  private ImageView myAccountAvatar;
+  private ImageView myStoreAvatar;
+  private TextView myAccountName;
+  private TextView myStoreName;
+  private Button loginButton;
+  private Button logoutButton;
+  private Button findFriendsButton;
+  private Button createStoreButton;
+  private Button editStoreButton;
+  private Button editProfileButton;
+
+  //Navigation buttons
+  private View notificationHistory;
+  private View settings;
+
+  //TODO: Add string resources to the settings XML and fragment (ALL STRINGS HARDCODED!!!)
+
+  public static Fragment newInstance() {
+    return new NewAccountFragment();
+  }
+
+  @Override public void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    getFragmentComponent(savedInstanceState).inject(this);
+
+    final AptoideApplication application =
+        (AptoideApplication) getContext().getApplicationContext();
+    accountManager =
+        ((AptoideApplication) getContext().getApplicationContext()).getAccountManager();
+
+    bodyInterceptor = application.getAccountSettingsBodyInterceptorPoolV7();
+    httpClient = application.getDefaultClient();
+    converterFactory = WebService.getDefaultConverter();
+  }
+
+  @Override public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
+    super.onViewCreated(view, savedInstanceState);
+
+    toolbar = (Toolbar) view.findViewById(R.id.toolbar);
+    notificationHistory = view.findViewById(R.id.notifications_history);
+    settings = view.findViewById(R.id.settings);
+
+    setAccountViews(view);
+    setupToolbar();
+
+    AptoideApplication application = (AptoideApplication) getContext().getApplicationContext();
+    attachPresenter(new NewAccountPresenter(this, accountManager, CrashReport.getInstance(),
+        application.getDefaultSharedPreferences(), AndroidSchedulers.mainThread(),
+        newAccountNavigator));
+  }
+
+  @Override public ScreenTagHistory getHistoryTracker() {
+    return ScreenTagHistory.Builder.build(this.getClass()
+        .getSimpleName());
+  }
+
+  @Override public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
+      @Nullable Bundle savedInstanceState) {
+    return inflater.inflate(R.layout.new_account_layout, container, false);
+  }
+
+  @Override public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String s) {
+
+  }
+
+  @Override public void showAccount(Account account) {
+    if (account.getNickname()
+        .isEmpty() || account.getNickname()
+        .equals(null) || account == null) {
```

Comment:
if account is null this throws an exception so no need account == null

Reply:
fixed